### PR TITLE
React to minimize request on XdgStackWorkspace

### PR DESCRIPTION
**Pull request by wineee**
need wlr_foreign_toplevel_manager/wlr_layer_shell for wlroots base dock/panal, cann't not test cancel minimize

### CODE REVIEW

File: src/server/kernel/wxdgsurface.cpp [hunk 1]
```diff
@@ -159,6 +161,13 @@ void WXdgSurfacePrivate::connect()
                 Q_EMIT q->requestToNormalState();
```
Comment (zccrs) [inline src/server/kernel/wxdgsurface.cpp:161]: rename to `requestCancelMaximize`

**Review by zccrs — CHANGES_REQUESTED** [review]

File: examples/tinywl/XdgStackWorkspace.qml [hunk 2]
```diff
@@ -154,6 +155,34 @@ Item {
...
+
+                function onRequestMinimize() {
+                    if (waylandSurface.isResizeing)
+                        return
+
+                    if (waylandSurface.isMinimize)
+                        return
+
+                    focus = false;
+                    Helper.setActivateSurface = null;
```
Comment (zccrs) [inline examples/tinywl/XdgStackWorkspace.qml:167]: `activeSurface`? And `if (Helper.activeSurface == surface)` before `Helper.activeSurface = null`

File: src/server/kernel/wxdgshell.h [hunk 3]
```diff
@@ -3,6 +3,7 @@
 
 #pragma once
 
+
```
Comment (zccrs) [inline src/server/kernel/wxdgshell.h:6]: ?

File: examples/tinywl/XdgStackWorkspace.qml [hunk 4]
```diff
@@ -154,6 +155,34 @@ Item {
...
+
+                    waylandSurface.setMinimize(true)
+                    surface.minimize = true;
+                }
+
+                function onRequestCancelMinimize() {
```
Comment (zccrs) [inline examples/tinywl/XdgStackWorkspace.qml:173]: How cancel minimize in this example?

**Review by zccrs — CHANGES_REQUESTED** [review]

File: examples/tinywl/XdgStackWorkspace.qml (hunk 4 above)
Comment (wineee) [reply examples/tinywl/XdgStackWorkspace.qml:173]: need wlr_foreign_toplevel_manager/wlr_layer_shell for wlroots base dock/panal, cann't not test cancel minimize now


This pr needs to hang for a while

Comment (zccrs) [reply examples/tinywl/XdgStackWorkspace.qml:173]: We can add a simple dock in this example.

File: examples/tinywl/XdgStackWorkspace.qml [hunk 5]
```diff
@@ -229,6 +302,38 @@ Item {
...
+                    if (waylandSurface.isResizeing)
+                        return
+
+                    if (!waylandSurface.isMinimized)
+                        return
+
+                    forceActiveFocus()
```
Comment (zccrs) [inline examples/tinywl/XdgStackWorkspace.qml:329]: Don't need

File: examples/tinywl/XdgStackWorkspace.qml [hunk 6]
```diff
@@ -229,6 +302,38 @@ Item {
...
+                    Helper.activatedSurface = waylandSurface
+
+                    visible = true;
+
+                    dockModel.removeSurface(surface)
+                    waylandSurface.setMinimize(false)
+                }
```
Comment (zccrs) [inline examples/tinywl/XdgStackWorkspace.qml:336]: If the state has been maximized before being minimized, it should be restored to its maximum state.

Comment (zccrs) [reply examples/tinywl/XdgStackWorkspace.qml:336]: Record the old state

File: examples/tinywl/XdgStackWorkspace.qml [hunk 7]
```diff
@@ -167,7 +235,12 @@ Item {
                 surface.outputCoordMapper = surface.CoordMapper.helper.get(output)
             }
 
+            function cancelMinimize () {
+                surfaceConnections.onRequestCancelMinimize();
```
Comment (zccrs) [inline examples/tinywl/XdgStackWorkspace.qml:239]: Move the inside codes of `onRequestCancelMinimize()` to here, and call `cancelMinimze()` in `onRequestCancelMinimize()`

**Review by zccrs — CHANGES_REQUESTED** [review]

**Review by zccrs — APPROVED** [review]

**Pull request merged**